src/client/mod.rs

```diff
@@ -784,22 +879,16 @@ impl<T> VaultClient<T>
     /// # fn main() {
     /// let host = "http://127.0.0.1:8200";
     /// let token = "test12345";
-    /// let client = Client::new(host, token).unwrap();
+    /// let client = Client::new(host, token, &[""]).unwrap();
     /// let res = client.set_secret("hello_set", "world");
     /// assert!(res.is_ok());
     /// # }
     /// ```
+    #[allow(unused_results)]
     pub fn set_secret<S1: Into<String>, S2: AsRef<str>>(&self, key: S1, value: S2) -> Result<()> {
-        let _ = try!(self.post::<_, String>(&format!("/v1/secret/{}", key.into())[..],
-                                            Some(&format!("{{\"value\": \"{}\"}}",
-                                                          self.escape(value.as_ref()))
-                                                      [..]),
-                                            None));
-        Ok(())
-    }
-
-    fn escape<S: AsRef<str>>(&self, input: S) -> String {
-        input.as_ref().replace("\n", "\\n")
+        let value = json!({"value": value.as_ref()}).to_string();
+        self.post::<_, String>(&format!("/v1/secret/{}", key.into()), Some(&value), None)
```

Comment:
I think you can use `let _ = self.post...?;` here and remove the `#[allow(unused_results)]`.